cleverpush/src/main/java/com/cleverpush/CleverPush.java

```diff
@@ -1692,146 +1754,146 @@ public void onFailure(int statusCode, String response, Throwable throwable) {
+                                editor.putString(CleverPushPreferences.SUBSCRIPTION_ATTRIBUTES, jsonString);
+                                editor.commit();
+                            }
+                        } catch (Exception ex) {
+                            Log.e("CleverPush", ex.getMessage(), ex);
+                        }
+                    }
+
+                    @Override
+                    public void onFailure(int statusCode, String response, Throwable throwable) {
+                        Log.e("CleverPush", "Error pushing attribute value - HTTP " + statusCode);
+                    }
+                });
+            }
+        })).start());
+    }
+
+    public void pullSubscriptionAttributeValue(String attributeId, String value) {
+        this.waitForTrackingConsent(() -> new Thread(() -> this.getSubscriptionId(subscriptionId -> {
+            if (subscriptionId != null) {
+                JSONObject jsonBody = new JSONObject();
+                try {
+                    jsonBody.put("channelId", this.channelId);
+                    jsonBody.put("attributeId", attributeId);
+                    jsonBody.put("value", value);
+                    jsonBody.put("subscriptionId", subscriptionId);
+                } catch (JSONException ex) {
+                    Log.e("CleverPush", ex.getMessage(), ex);
+                }
+
+                SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(CleverPush.context);
+                Map<String, Object> subscriptionAttributes = this.getSubscriptionAttributes();
+
+                ArrayList<String> arrayList = new ArrayList<>();
+                try {
+                    JSONArray arrayValue = (JSONArray) subscriptionAttributes.get(attributeId);
+                    if (arrayValue != null) {
+                        for (int i = 0; i < arrayValue.length(); i++) {
+                            if (!arrayValue.getString(i).equals(value)) {
+                                arrayList.add(arrayValue.getString(i));
+                            }
+                        }
+                    }
+                } catch (Exception ex) {
+                    Log.e("CleverPush", ex.getMessage(), ex);
+                }
+
+                String[] arrayString = arrayList.toArray(new String[0]);
+                if (arrayString == null) {
+                    return;
+                }
+                subscriptionAttributes.put(attributeId, arrayString);
+
+                CleverPushHttpClient.post("/subscription/attribute/pull-value", jsonBody, new CleverPushHttpClient.ResponseHandler() {
+                    @Override
+                    public void onSuccess(String response) {
+                        try {
+                            if (sharedPreferences != null) {
+                                JSONObject jsonObject = new JSONObject(subscriptionAttributes);
```

Comment:
is this ResponseHandler supposed to do the same as /subscription/attribute/push-value? if so, please use the exported ResponseHandler